### PR TITLE
WIP: Feature/7.4 alpine images

### PR DESCRIPTION
**Pull request by DracoBlue**
- [x] add alpine version of the images
- [x] upgrade pecl extensions (memcached) for non-alpine image
- [x] integrate quay.io to see CVEs https://quay.io/repository/exozet/php-fpm?tab=tags
Bildschirmfoto 2019-12-15 um 08 53 27
- [x] compare -i of non-alpine and xdebug-alpine version. here are the differences (one by one handpicked!)

[7.4.0-xdebug-alpine.i-report.txt](https://github.com/exozet/docker-php-fpm/files/3964717/7.4.0-xdebug-alpine.i-report.txt):

        - librabbitmq version => 0.8.0
        - log_errors => On => On
        - cURL Information => 7.66.0
            - IDN => No
            - SPNEGO => No
            - GSSAPI => No
            - KERBEROS5 => No
            - PSL => No
            - Protocols = dict, file, ftp, ftps, gopher, http, https, imap, imaps, pop3, pop3s, rtsp, smb, smbs, smtp, smtps, telnet, tftp
        - libxml Version => 2.9.9
        - gd
            - FreeType Version => 2.10.0
            - libJPEG Version => 8
            - libPNG Version => 1.6.37
         - iconv
            - iconv implementation => unknown
            - iconv library version => unknown
         - imagick
            - Imagick compiled with ImageMagick version => ImageMagick 7.0.8-58
            - ImageMagick release date => 2019-08-13
            - ImageMagick number of supported formats:  => 231
            - ImageMagick supported formats => 3FR, 3G2, 3GP, AAI, AI, ART, ARW, AVI, AVS, BGR, BGRA, BGRO, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CRW, CUBE, CUR, CUT, DATA, DCM, DCR, DCRAW, DCX, DDS, DFONT, DNG, DOT, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, FAX, FILE, FITS, FLV, FRACTAL, FTP, FTS, G3, G4, GIF, GIF87, GRADIENT, GRAY, GRAYA, GROUP4, GV, HALD, HDR, HEIC, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, JNG, JNX, JPE, JPEG, JPG, JPS, JSON, K25, KDC, LABEL, M2V, M4V, MAC, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPEG, MPG, MRW, MSL, MSVG, MTV, MVG, NEF, NRW, NULL, ORF, OTB, OTF, PAL, PALM, PAM, PANGO, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, PPM, PS, PS2, PS3, PSB, PSD, PTIF, PWP, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGB565, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, SUN, SVG, SVGZ, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TM2, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIFF, VIPS, VST, WBMP, WEBP, WMV, WPG, X, X3F, XBM, XC, XCF, XPM, XPS, XV, XWD, YCbCr, YCbCrA, YUV
          - intl
            - ICU version => 64.2
          - ldap
            - Vendor Version => 20448
          - libxml
             - libXML Compiled Version => 2.9.9
             - libXML Loaded Version => 20909
           - mbstring
             - Multibyte regex (oniguruma) version => 6.9.3
           - PostgreSQL(libpq) Version => 11.6
           - SQLite Library => 3.28.0
           - PostgreSQL(libpq) Version => 11.6
           - PostgreSQL(libpq)  => PostgreSQL 11.6
           - libsodium headers version => 1.0.18
           - SQLite Library => 3.28.0
           - libxml2 Version => 2.9.9
           - libxslt Version => 1.1.33
           - Libzip headers version => 1.5.2

[7.4.0.i-report.txt](https://github.com/exozet/docker-php-fpm/files/3964718/7.4.0.i-report.txt):

        - librabbitmq version => 0.9.0
        - log_errors => Off => Off
        - cURL Information => 7.64.0
            - IDN => Yes
            - SPNEGO => Yes
            - GSSAPI => Yes
            - KERBEROS5 => Yes
            - PSL => Yes
            - Protocols => dict, file, ftp, ftps, gopher, http, https, imap, imaps, ldap, ldaps, pop3, pop3s, rtmp, rtsp, scp, sftp, smb, smbs, smtp, smtps, telnet, tftp
         - libxml Version => 2.9.4
        - gd
            - FreeType Version => 2.9.1
            - libJPEG Version => 6b
            - libPNG Version => 1.6.36
         - iconv
            - iconv implementation => glibc
            - iconv library version => 2.28
         - imagick
            - Imagick compiled with ImageMagick version => ImageMagick  6.9.10-23
            - ImageMagick release date => 20190101
            - ImageMagick number of supported formats:  => 242
            - ImageMagick supported formats => 3FR, 3G2, 3GP, AAI, AI, ART, ARW, AVI, AVS, BGR, BGRA, BGRO, BIE, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CRW, CUR, CUT, DATA, DCM, DCR, DCX, DDS, DFONT, DJVU, DNG, DOT, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, EXR, FAX, FILE, FITS, FRACTAL, FTP, FTS, G3, G4, GIF, GIF87, GRADIENT, GRAY, GRAYA, GROUP4, GV, H, HALD, HDR, HEIC, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, J2C, J2K, JBG, JBIG, JNG, JNX, JP2, JPC, JPE, JPEG, JPG, JPM, JPS, JPT, JSON, K25, KDC, LABEL, M2V, M4V, MAC, MAGICK, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPEG, MPG, MRW, MSL, MSVG, MTV, MVG, NEF, NRW, NULL, ORF, OTB, OTF, PAL, PALM, PAM, PANGO, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, PPM, PREVIEW, PS, PS2, PS3, PSB, PSD, PTIF, PWP, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, SUN, SVG, SVGZ, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIFF, VIPS, VST, WBMP, WEBP, WMF, WMV, WMZ, WPG, X, X3F, XBM, XC, XCF, XPM, XPS, XV, XWD, YCbCr, YCbCrA, YUV
          - intl
            - ICU version => 63.1
          - ldap
            - Vendor Version => 20447
          - libxml
             - libXML Compiled Version => 2.9.4
             - libXML Loaded Version => 20904
           - mbstring
             - Multibyte regex (oniguruma) version => 6.9.1
           - PostgreSQL(libpq) Version => 11.5 (Debian 11.5-1+deb10u1)
           - SQLite Library => 3.27.2
           - PostgreSQL(libpq) Version => 11.5
           - PostgreSQL(libpq)  => PostgreSQL 11.5
           - libsodium headers version => 1.0.17
           - SQLite Library => 3.27.2
           - libxml2 Version => 2.9.4
           - libxslt Version => 1.1.32
           - Libzip headers version => 1.5.1

### CODE REVIEW

File: 7.4/Dockerfile-alpine [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+FROM build-base AS build-xdebug
+RUN pecl install xdebug
+RUN docker-php-ext-enable xdebug
+RUN cp `php -i | grep ^extension_dir | cut -f 3 -d ' '`/*.so /tmp/
+
+FROM base
+
+ENV COMPOSER_HOME=/composer
+COPY --from=composer:1.8.6 /usr/bin/composer /usr/local/bin/composer
```
Comment (OskarStark) [inline 7.4/Dockerfile-alpine]: 1.9.1

File: 7.4/Dockerfile-alpine [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    PHP_UPLOAD_MAX_FILESIZE=2M \
+    PHP_XDEBUG_DEFAULT_ENABLE=0 \
+    PHP_XDEBUG_IDEKEY=''\
+    PHP_XDEBUG_PROFILER_ENABLE=0 \
+    PHP_XDEBUG_REMOTE_AUTOSTART=0 \
+    PHP_XDEBUG_REMOTE_CONNECT_BACK=0 \
+    PHP_XDEBUG_REMOTE_ENABLE=0 \
+    PHP_XDEBUG_REMOTE_HOST=0
+
+WORKDIR /usr/src/app
```
Comment (OskarStark) [inline 7.4/Dockerfile-alpine]: Missing newline

File: start-cron-alpine [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# trap SIGINT and SIGTERM signals and gracefully exit
+trap "service cron stop; kill \$!; exit" SIGINT SIGTERM
+
+# start "daemon"
+while true
+do
+    # watch /var/log/cron.log restarting if necessary
+    cat /var/log/cron.log & wait $!
+done
```
Comment (OskarStark) [inline start-cron-alpine]: Missing newline

**Comment by DracoBlue** [comment]
> I like the PR, but it’s quite variable if we cannot (?) specify Version numbers 🧐

version numbers are possible for all extensions. I was just trying to get it work with 7.4 :)

File: 7.4/Dockerfile-alpine (hunk 1 above)
Comment (DracoBlue) [reply 7.4/Dockerfile-alpine]: please fix it upstream. it's still 1.8.6 in the non-alpine images, too.

**Comment by DracoBlue** [comment]
@OskarStark versions pinned for pecl packages with 1589536

**Comment by OskarStark** [comment]
Great work 👍🏻

**Comment by OskarStark** [comment]
Now we should switch to Github Actions for building

**Pull request merged**